modules/datadict/php/datadict.class.inc

```diff
@@ -113,16 +113,149 @@ class Datadict extends \NDB_Menu_Filter
+    {
+        // 1. Reset filters
+        if (isset($_REQUEST['reset'])) {
+            // erase filter data from the session
+            $this->_resetFilters(true);
+        }
+
+        // 2. After resetting filters (if applicable), set the new
+        // filters using the appropriate method.
+        if ($_SERVER['REQUEST_METHOD'] === 'GET'
+            || ($_SERVER['REQUEST_METHOD'] === 'POST' && !isset($_POST['filter']))
+        ) {
+            // If it's a GET request (ie just reloading the page with
+            // previous filters or clicking on a sort column, use
+            // the filters that were saved into the session.
+            // OR a post request with no new defined filters (ie a second
+            // form on the page)
+            $savedFilters = $_SESSION['State']->getProperty('filter');
+            $this->_setFilters($savedFilters);
+
+            // Also set the form defaults based on the saved query, so that
+            // the filters values remain populated correctly in the front-end,
+            // not just in the query..
+            $defaults = $this->_getDefaults();
+            if (is_array($savedFilters)) {
+                $this->_setDefaults(array_merge($defaults, $savedFilters));
+            } else {
+                $this->_setDefaults($defaults);
+            }
+        } else if ($_SERVER['REQUEST_METHOD'] === 'POST'
+            && isset($_POST['filter'])
+        ) {
+            // If it's a POST, set the filters based on the
+            // request and save them into the session so that future
+            // GET requests have access to them.
+            $newFilters = $this->_setFilters($_POST);
+            if (!empty($newFilters)) {
+                $_SESSION['State']->setProperty('filter', $newFilters);
+            }
+        }
+
+        // 3. Set the search keyword (if applicable)
+        $key = null;
+        if (isset($_REQUEST['keyword'])) {
+            // It was included in the request, so save it.
+            $key = $_REQUEST['keyword'];
+            // Check if any quoted keyword
+            if (preg_match_all(
+                '\"[\w\s]+\"|\\\'[\w\s]+\\\'|[\w\s]+',
+                $key,
+                $keywords
+            )
+            ) {
+                $keywords = array_filter(
+                    array_map(
+                        function ($item) {
+                            return trim($item, ' ');
+                        },
+                        $keywords[0]
+                    )
+                );
+            }
+            if (is_array($keywords)) {
+                // Quoted keyword(s) exist.
+                $keywords_extra = array();
+                foreach ($keywords as &$keyword) {
+                    if (preg_match('\"[\w\s]+\"', $keyword)) {
+                        // Handle the quoted keyword.
+                        $keyword = '(?i)^' . $keyword . '$';
+                    } else if (preg_match('/\\s/', $keyword)) {
```

Comment:
I think you are missing an `if` here to handle the case where `$keyword` is delimited by single quotes.

Reply:
You're right. 

I thought it would be best to not include single quotes. The code logic of splitting the segments into an array would likely break if a user had for example:
'I'm 27 years old'

What do you think is best?

Reply:
I think single and double quotes should be supported. You can either add an `if` clause here or modify the regex in your first `if` test so that it supports both quoting options.

Reply:
I'm not necessarily a regular expression expert. I noticed the problem @davidblader mentions is because the regexp I created needs fixing to include parenthesis with other characters. 

Currently the regexp code I wrote will take a search as example:
"exact search example" independent word searches "another exact search" ..

The result will do searches for exact two times and independent word searches.

The reason I'm leaning towards removing single quotes support is because of cases where single quotes are used multiple times and such as in the description column. I think it's unnecessary with added complexity but i'll take a look if I can figure out some magic with regex.